supervision/dataset/classification.py

```diff
@@ -0,0 +1,239 @@
+import os
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple
+
+import cv2
+import numpy as np
+
+from supervision.dataset.core import BaseDataset
+from supervision.dataset.ultils import train_test_split
+
+
+def _validate_class_ids(class_id: Any, n: int) -> None:
+    """
+    Ensure that class_id is a 1d np.ndarray with (n, ) shape.
+    """
+    is_valid = isinstance(class_id, np.ndarray) and class_id.shape == (n,)
+    if not is_valid:
+        raise ValueError("class_id must be 1d np.ndarray with (n, ) shape")
+
+
+def _validate_confidence(confidence: Any, n: int) -> None:
+    """
+    Ensure that confidence is a 1d np.ndarray with (n, ) shape.
+    """
+    is_valid = isinstance(confidence, np.ndarray) and confidence.shape == (n,)
+    if not is_valid:
+        raise ValueError("confidence must be 1d np.ndarray with (n, ) shape")
+
+
+@dataclass
+class Classifications:
```

Comment:
This class should not live in the `dataset` package.

Reply:
Where should it be?

Reply:
I'd probably create a new file `supervision.classification.core` and move that part of the code there.